Source/WebCore/Modules/webcodecs/WebCodecsAudioDecoder.cpp

```diff
@@ -150,24 +151,24 @@ ExceptionOr<void> WebCodecsAudioDecoder::decode(Ref<WebCodecsEncodedAudioChunk>&
     if (m_state != WebCodecsCodecState::Configured)
         return Exception { ExceptionCode::InvalidStateError, "AudioDecoder is not configured"_s };
 
+    Ref pendingActivityKeepingJSWrapperAlive = makePendingActivity(*this);
+
     if (m_isKeyChunkRequired) {
         if (chunk->type() != WebCodecsEncodedAudioChunkType::Key)
             return Exception { ExceptionCode::DataError, "Key frame is required"_s };
         m_isKeyChunkRequired = false;
     }
 
     ++m_decodeQueueSize;
-    queueControlMessageAndProcess([this, chunk = WTFMove(chunk)]() mutable {
-        ++m_beingDecodedQueueSize;
+    queueControlMessageAndProcess({ *this, [this, chunk = WTFMove(chunk)]() mutable {
         --m_decodeQueueSize;
         scheduleDequeueEvent();
 
-        m_internalDecoder->decode({ chunk->span(), chunk->type() == WebCodecsEncodedAudioChunkType::Key, chunk->timestamp(), chunk->duration() }, [this](String&& result) {
-            --m_beingDecodedQueueSize;
+        m_internalDecoder->decode({ chunk->span(), chunk->type() == WebCodecsEncodedAudioChunkType::Key, chunk->timestamp(), chunk->duration() }, [this, pendingActivity = makePendingActivity(*this)](String&& result) {
```

Comment:
This sounds good to use pending activity instead of m_beingDecodedQueueSize here.

Reply:
Yeah this works well to cut down on the number of state variables. Now the codecs only have the spec-defined variables and the `m_clearFlushPromiseCount` for synchronizing flushes. (And `m_isKeyChunkRequired` which doesn't appear to actually be used anymore?)